tests/float/float_dunder.py

```diff
@@ -0,0 +1,49 @@
+# test __float__ function support
+
+
+class TestFloat:
+    def __float__(self):
+        return 10.0
+
+
+class TestStrFloat:
+    def __float__(self):
+        return "a"
+
+
+class TestNonFloat:
+    def __float__(self):
+        return 6
+
+
+class Test:
+    pass
+
+
+# try:
+#     # implicit conversion to float should fail - doesn't on mpy
+#     2.0 * TestFloat()
```

Comment:
Found a new problem - this behaviour does not match cpython, neither here nor on the previous implementation.